contents/data/sql/SQL.md

```diff
@@ -0,0 +1,242 @@
+
+##### 3. `GROUP BY Course`
+
+This is used to group the result set, and is often used with aggregate functions (like `COUNT`, `MAX`, `AVG` etc.).
+
+In this query, we are essentially grouping into 3 "groups": `CS101` group (IDs 5 and 7), `CS202` group (IDs  1, 3, and 4), and `CS404` group (ID 8).
+
+##### 4. `HAVING COUNT(*) > 1`
+This is a conditional clause that is used with aggregate functions. `COUNT(*)` is an aggregate function that returns the number of rows in each "group".
+
+So, this clause will filter out the `CS404` group since it has a count of 1, and doesn't satisfy the condition.
+
+##### 5. `SELECT Course, COUNT(*) num`
+The `SELECT` statement is used for choosing the data to display. The returned data is stored in a result table, called the **result-set**.
+
+In this statement, we choose the `Course` and its count, `num` to be the result-set that is to be displayed. Adding `num` after `COUNT(*)` means that it is an alias for `COUNT(*)`. The result-set will look like this:
+
+| Course | num |
+| ------ | --- |
+| CS202  | 3   |
+| CS101  | 2   |
+Table 4. Result-set after `SELECT`
+
+<box type="info">
+  Since, we have used a <code>GROUP BY</code> clause, we cannot <code>SELECT</code> data from an individual attribute (like ID or Name) that is not part of the aggregate data generated in the group (like Course).
+</box>
+
+##### 6. `ORDER BY num`
+
+Finally, we need to sort our result entries. We use the `ORDER BY` clause (which sorts in ascending order by default) for doing so.
+
+In this clause, we order the result-set in ascending order based on the `num` attribute (column). So, our final output is:
+
+| Course | num |
+| ------ | --- |
+| CS101  | 2   |
+| CS202  | 3   |
+Table 4. Final result after `ORDER BY`
+
+
+This is the same as our expected output from Table 2!
+
+<box type="info">
+  You can experiment with this example on <a href="https://www.db-fiddle.com/f/yxjjgbkKmsa46cKjeEg1X/2">DB Fiddle</a> by entering queries into the <code>Query SQL</code> pane and then clicking the <code>Run</code> icon.
+</box>
+
+The example shown here is relatively simple. Typical SQL queries have the capability to be much more complex as there are a lot of clauses, functions, and operators that are not covered here. As the complexity of the query grows, there is a possible decrease in the performance of the query. This is why [query planning](https://www.khanacademy.org/computing/computer-programming/sql/relational-queries-in-sql/a/more-efficient-sql-with-query-planning-and-optimization) and [optimization](https://www.sisense.com/blog/8-ways-fine-tune-sql-queries-production-databases/) is important.
+
+---
+
+## Why learn SQL?
+
+Being open source and one of the first DBMSs, MySQL (a specific implementation of SQL) increased the popularity of **SQL** in its early days. It ended up being a core component of the <tooltip content="Linux OS, Apache Server, MySQL RDBMS, PHP  language">LAMP</tooltip> web application stack (and many others).
```

Comment:
What does this sentence have to do with 'Why learn SQL?'

Reply:
It provides some context for `increased the popularity of SQL in its early days`.

I guess I can remove it too